gtk4/src/subclass/cell_layout.rs

```diff
@@ -0,0 +1,368 @@
+    let instance = &*(cell_layout as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    imp.clear(from_glib_borrow::<_, CellLayout>(cell_layout).unsafe_cast_ref())
+}
+
+unsafe extern "C" fn cell_layout_reorder<T: CellLayoutImpl>(
+    cell_layout: *mut ffi::GtkCellLayout,
+    cellptr: *mut ffi::GtkCellRenderer,
+    position: i32,
+) {
+    let instance = &*(cell_layout as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    let cell: Borrowed<CellRenderer> = from_glib_borrow(cellptr);
+
+    imp.reorder(
+        from_glib_borrow::<_, CellLayout>(cell_layout).unsafe_cast_ref(),
+        &cell,
+        position,
+    )
+}
+
+unsafe extern "C" fn cell_layout_add_attribute<T: CellLayoutImpl>(
+    cell_layout: *mut ffi::GtkCellLayout,
+    cellptr: *mut ffi::GtkCellRenderer,
+    attributeptr: *const libc::c_char,
+    column: i32,
+) {
+    let instance = &*(cell_layout as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    let cell: Borrowed<CellRenderer> = from_glib_borrow(cellptr);
+    let attribute: Borrowed<glib::GString> = from_glib_borrow(attributeptr);
+
+    imp.add_attribute(
+        from_glib_borrow::<_, CellLayout>(cell_layout).unsafe_cast_ref(),
+        &cell,
+        &attribute,
+        column,
+    )
+}
+
+unsafe extern "C" fn cell_layout_clear_attributes<T: CellLayoutImpl>(
+    cell_layout: *mut ffi::GtkCellLayout,
+    cellptr: *mut ffi::GtkCellRenderer,
+) {
+    let instance = &*(cell_layout as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    let cell: Borrowed<CellRenderer> = from_glib_borrow(cellptr);
+
+    imp.clear_attributes(
+        from_glib_borrow::<_, CellLayout>(cell_layout).unsafe_cast_ref(),
+        &cell,
+    )
+}
+
+unsafe extern "C" fn cell_layout_set_cell_data_func<T: CellLayoutImpl>(
+    cell_layout: *mut ffi::GtkCellLayout,
+    cellptr: *mut ffi::GtkCellRenderer,
+    callback: ffi::GtkCellLayoutDataFunc,
+    user_data: glib::ffi::gpointer,
+    destroy_notify: glib::ffi::GDestroyNotify,
+) {
+    let instance = &*(cell_layout as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    let cell: Borrowed<CellRenderer> = from_glib_borrow(cellptr);
+
+    let callback = if callback.is_none() {
+        None
+    } else {
+        Some(CellLayoutDataCallback {
+            callback,
+            user_data,
+            destroy_notify,
+        })
+    };
+
+    imp.set_cell_data_func(
+        from_glib_borrow::<_, CellLayout>(cell_layout).unsafe_cast_ref(),
+        &cell,
+        callback.as_ref(),
+    )
+}
+
+unsafe extern "C" fn cell_layout_get_cells<T: CellLayoutImpl>(
+    cell_layout: *mut ffi::GtkCellLayout,
+) -> *mut glib::ffi::GList {
+    let instance = &*(cell_layout as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    let cells = imp.get_cells(from_glib_borrow::<_, CellLayout>(cell_layout).unsafe_cast_ref());
+    ToGlibContainerFromSlice::to_glib_container_from_slice(&cells).0
```

Comment:
You need to keep the cells themselves alive, they're `transfer none` and only the container is owned by the caller

Reply:
I suppose storing the cells on a qdata is enough to keep them alive? :)

Reply:
I guess so